src/main/java/org/wso2/carbon/connector/operations/DeleteFileOrFolder.java

```diff
@@ -62,79 +62,117 @@ public void connect(MessageContext messageContext) throws ConnectException {
+                fileOrFolderPath = fileSystemHandlerConnection.getBaseDirectoryPath() + fileOrFolderPath;
+                fileObjectToDelete = fsManager.resolveFile(fileOrFolderPath, fso);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Delete file/folder attempt " + attempt + " of " + maxRetries + " for file/folder "
+                            + fileOrFolderPath);
                 }
-                isOperationSuccessful = true;
-                OMElement numOfDeletedFilesEle = Utils.
-                        createOMElement(NUM_OF_DELETED_FILES_ELE,
-                                Integer.toString(numberOfDeletedFiles));
-                result = new FileOperationResult(
-                        OPERATION_NAME,
-                        isOperationSuccessful,
-                        numOfDeletedFilesEle);
-            }
-
-            Utils.setResultAsPayload(messageContext, result);
 
-        } catch (InvalidConfigurationException e) {
+                //Deletes this file. Does nothing if this file does not exist
+                if (fileObjectToDelete.isFile()) {
+                    isOperationSuccessful = fileObjectToDelete.delete();
+                    result = new FileOperationResult(
+                            OPERATION_NAME,
+                            isOperationSuccessful);
+                }
 
-            String errorDetail = ERROR_MESSAGE + fileOrFolderPath;
-            handleError(messageContext, e, Error.INVALID_CONFIGURATION, errorDetail);
+                if (fileObjectToDelete.isFolder()) {
+                    int numberOfDeletedFiles;
+                    if (StringUtils.isNotEmpty(fileMatchingPattern)) {
+                        FileFilter fileFilter = new SimpleFileFiler(fileMatchingPattern);
+                        FileFilterSelector fileFilterSelector = new FileFilterSelector(fileFilter);
+                        /*
+                         * Deletes all descendants of this file that
+                         * match a selector. Does nothing if this
+                         * file does not exist.
+                         */
+                        numberOfDeletedFiles = fileObjectToDelete.delete(fileFilterSelector);
+                    } else {
+                        //Deletes this file and all children.
+                        numberOfDeletedFiles = fileObjectToDelete.deleteAll();
+                    }
+                    isOperationSuccessful = true;
+                    OMElement numOfDeletedFilesEle = Utils.
+                            createOMElement(NUM_OF_DELETED_FILES_ELE,
+                                    Integer.toString(numberOfDeletedFiles));
+                    result = new FileOperationResult(
+                            OPERATION_NAME,
+                            isOperationSuccessful,
+                            numOfDeletedFilesEle);
+                }
 
-        } catch (FileSystemException e) {
```

Comment:
we need to add the retry logic to the FileSystemException as it usually contain the file connection exceptions

Reply:
Since we do the retry for general exception this will be covered from it